stetho-repl-js/src/main/java/com/facebook/stetho/inspector/console/js/JavaScriptRuntimeBuilder.java

```diff
@@ -0,0 +1,248 @@
+
+    ScriptableObject.putProperty(scope, "context", Context.javaToJS(mContext, scope));
+
+    // Import the classes that the caller requested
+    for (Class<?> aClass : mClasses) {
+      String className = aClass.getName();
+      try {
+        String expression = String.format("importClass(%s)", className);
+        LogUtil.i("js) init import class: %s", expression);
+        jsContext.evaluateString(scope, expression, "chrome", 1, null);
+      } catch (Exception e) {
+        LogUtil.e(e, "js) Failed to import class: %s", className);
+        String message = String.format("Failed to import class: %s; %s", className, Log.getStackTraceString(e));
+        CLog.writeToConsole(Console.MessageLevel.ERROR, Console.MessageSource.JAVASCRIPT, message);
+      }
+    }
+
+    // Import the packages that the caller requested
+    for (String packageName : mPackages) {
+      try {
+        String expression = String.format("importPackage(%s)", packageName);
+        LogUtil.i("js) init import package: %s", expression);
+        jsContext.evaluateString(scope, expression, "chrome", 1, null);
+      } catch (Exception e) {
+        LogUtil.e(e, "js) Failed to import package: %s", packageName);
+        String message = String.format("Failed to import package: %s; %s", packageName, Log.getStackTraceString(e));
+        CLog.writeToConsole(Console.MessageLevel.ERROR, Console.MessageSource.JAVASCRIPT, message);
+      }
+    }
+
+    // Set the `console` object
+    try {
+      ScriptableObject.defineClass(scope, JavaScriptRuntimeConsole.class);
+      JavaScriptRuntimeConsole console = new JavaScriptRuntimeConsole(scope);
+      LogUtil.i("js) init add console");
+      scope.defineProperty("console", console, ScriptableObject.DONTENUM);
+    } catch (Exception e) {
+      LogUtil.e(e, "js) Failed to setup javascript console");
+      CLog.writeToConsole(Console.MessageLevel.ERROR, Console.MessageSource.JAVASCRIPT, "Failed to setup javascript console");
+    }
+
+    // Define the variables
+    for (Map.Entry<String, Object> entrySet : variables.entrySet()) {
+      String varName = entrySet.getKey();
+      Object varValue = entrySet.getValue();
+      try {
+        Object jsValue = Context.javaToJS(varValue, scope);
+        ScriptableObject.putProperty(scope, varName, jsValue);
+      } catch (Exception e) {
+        LogUtil.e(e, "js) Failed to setup javascript variable: %s", varName);
+        String message = String.format("Failed to setup javascript variable: %s", varName, Log.getStackTraceString(e));
+        CLog.writeToConsole(Console.MessageLevel.ERROR, Console.MessageSource.JAVASCRIPT, message);
+      }
+    }
+
+    // Define the functions
+    for (Map.Entry<String, Function> entrySet : functions.entrySet()) {
+      String functionName = entrySet.getKey();
+      Function function = entrySet.getValue();
+      try {
+        ScriptableObject.putProperty(scope, functionName, function);
+      } catch (Exception e) {
```

Comment:
One thing you can do to centralize some of the exception handling as well is break this up into separate functions so the call stack includes part of the operation being carried out.  You can also do a simple custom exception class to centralize the flow control as with this simple example:

```java
void centralFunction() {
  try {
    doFoo();
    doBar();
    doBaz();
  } catch (StethoJSException e) {
    Throwable realException = e.getCause();
    String message = e.getMessage();
    jsCLog(message, realException);
  }
}

void doFoo() throws StethoJSException {
  try {
    thisCrazyRhinoThing();
  } catch (Exception e) {
    throw new StethoJSException("Error doing this crazy Rhino thing", e);
  }
}
```